cocos/scene-graph/node.jsb.ts

```diff
@@ -1214,11 +1214,25 @@ nodeProto[serializeTag] = function (serializationOutput: SerializationOutput, co
         // discard props disallow to synchronize
         const isRoot = this._prefab?.root === this;
         if (isRoot) {
-            serializationOutput.writeProperty('_objFlags', this._objFlags);
-            serializationOutput.writeProperty('_parent', this._parent);
-            serializationOutput.writeProperty('_prefab', this._prefab);
-            if (context.customArguments.keepNodeUuid) {
-                serializationOutput.writeProperty('_id', this._id);
+            // if B prefab is in A prefab,B can be referenced by component.We should discard it.because B is not the root of prefab
+            let isNestedPrefab = false;
+            let parent = this.getParent();
+            while (parent) {
+                const nestedRoots = parent?._prefab?.nestedPrefabInstanceRoots;
```

Comment:
I think `parent?` can change to `parent` as it is checked in while condition.

Reply:
done